tasks/eva_2429/migration_util.py

```diff
@@ -0,0 +1,72 @@
+import os.path
+import subprocess
+
+from ebi_eva_common_pyutils.command_utils import run_command_with_output
+from ebi_eva_common_pyutils.logger import logging_config
+
+logger = logging_config.get_logger(__name__)
+
+
+def export_data(mongo_source, accession_export_directory, mongoexport_args=None):
+    mongoexport_args = " ".join([f"--{arg} {val}"
+                                 for arg, val in mongoexport_args.items()]) if mongoexport_args else ""
+    mongoexport_command = f"mongoexport --uri {mongo_source.uri_with_db_name}  --out {accession_export_directory} {mongoexport_args}" + \
+                          mongo_source._get_optional_secrets_file_stdin()
+    try:
+        run_command_with_output("mongoexport", mongoexport_command, log_error_stream_to_output=True)
+    except subprocess.CalledProcessError as ex:
+        raise Exception("mongoexport failed! HINT: Did you forget to provide a secrets file for authentication?")
+
+
+def import_data(mongo_dest, coll_file_loc, mongoimport_args=None):
+    mongoimport_args = " ".join([f"--{arg} {val}"
+                                 for arg, val in mongoimport_args.items()]) if mongoimport_args else ""
+    mongoimport_command = f"mongoimport --uri {mongo_dest.uri_with_db_name}  --file {coll_file_loc} {mongoimport_args}" + \
+                          mongo_dest._get_optional_secrets_file_stdin()
+    try:
+        run_command_with_output("mongoimport", mongoimport_command, log_error_stream_to_output=True)
+    except subprocess.CalledProcessError as ex:
+        raise Exception("mongoexport failed! HINT: Did you forget to provide a secrets file for authentication?")
+
+
+def mongo_import_from_dir(mongo_dest, export_dir):
+    mongo_args = {
+        "mode": "upsert"
+    }
+    db_list = os.listdir(export_dir)
+
+    for db in db_list:
+        invalidate_and_set_db(mongo_dest, db)
+        db_dir = os.path.join(export_dir, db)
+        all_coll_dir = os.listdir(db_dir)
+        for coll in all_coll_dir:
+            logger.info(f'Importing data for db ({db} - collection ({coll})')
+            coll_dir = os.path.join(db_dir, coll)
+            files_list = os.listdir(coll_dir)
+            for file in files_list:
+                mongo_args.update({"collection": coll})
+                import_data(mongo_dest, os.path.join(coll_dir, file), mongo_args)
+
+
+def invalidate_and_set_db(mongo_instance, db):
+    if 'uri_with_db_name' in mongo_instance.__dict__:
+        del mongo_instance.__dict__['uri_with_db_name']
+    mongo_instance.db_name = db
```

Comment:
If this is a common use case we should change the `MongoDatabase` to allow the database to be reset. 
Although the name of the class imply that it is set for one database so maybe we should create one object per database.

Reply:
we can change the db in mongoDatabase but the property because it is cached does not change and that's why we have to invalidate it whenever we change the db. 
I think working with a single object is better than creating one for every db (taking cue from java), since it is just a matter of setting the db parameter in it. @sundarvenkata-EBI  ??

Reply:
@nitin-ebi I see your point about connection pooling and using a single instance. But in this sort of one-off script executions, it might be overkill to reuse connections by mutating the variable. We can attempt modeling such connection pooling at a driver level for long-lived scripts (you can create a ticket to investigate this) later but for now, I second Tim's suggestion to create separate connections.

Reply:
done